Feature/rafael sapalo/#57/create journal microservice

Rafael-Sapalo opened the pull request.

Rafael-Sapalo committed:
feat: add journal microservice with basic Hono server setup

- Created journal service with Hono framework and class-based architecture
- Added JournalService class with /journal base path and /health endpoint
- Configured Bun server to use JOURNAL_SERVICE_PORT environment variable
- Added dependencies: hono, drizzle-orm, pg for database integration
- Added dev dependencies: drizzle-kit, tsx, @types/bun, @types/pg
- Created .gitignore to exclude node_modules, env files, logs, and IDE configurations

Rafael-Sapalo committed:
feat: add database configuration and schema for journal service

- Created db.config.ts with Drizzle ORM PostgreSQL configuration supporting both development and production environments
- Added getDatabaseConfig function with environment variable support and production validation
- Created journal_entries table schema with id, userId, title, content, mood, and timestamps
- Created journal_tags table for user-defined tags with id, userId, and name fields
- Created journal_entry_tags junction table for many

Rafael-Sapalo committed:
feat: add journal service API endpoint placeholders for CRUD operations

- Added GET endpoints for retrieving tags, journal entries, and entry by ID
- Added POST endpoints for creating tags, journal entries, and adding media to entries
- Added PUT endpoints for updating journal entries and tags by ID
- Added DELETE endpoints for removing journal entries, tags, and entry media
- All endpoints return placeholder text responses for future implementation

Rafael-Sapalo committed:
feat: add controller structure and reorganize journal service routes

- Created JournalController class placeholder for journal entry operations
- Created empty media.controller.ts and tags.controller.ts files for future implementation
- Reorganized journal service routes by grouping entries, media, and tags endpoints
- Reordered routes to improve logical flow: entries first, then media operations, then tags

Rafael-Sapalo committed:
feat: add database indexes and implement journal entry controller handlers

- Added index imports and created indexes on userId fields for journal_entries, journal_tags, and journal_entry_tags tables
- Added indexes on entryId and tagId fields in journal_entry_tags and journal_media tables for improved query performance
- Added unique and notNull constraints to journal_entries id field
- Implemented JournalController handler methods: getEntries, getEntryById, createEntry, updateEntry, deleteEntry
-

Rafael-Sapalo committed:
feat: implement JournalRepository with database queries and integrate with controller handlers

- Created JournalRepository class with database query methods using Drizzle ORM
- Added getEntries method to fetch all journal entries from database
- Added getEntryById method to fetch single entry by ID with null handling
- Added createEntry method with transaction support, validation for userId and title, and error handling
- Added placeholder updateEntry and deleteEntry methods for future implementation
-

Rafael-Sapalo committed:
feat: add environment validation and npm scripts to journal service

- Added JOURNAL_SERVICE_PORT environment variable validation with error handling and process exit on startup
- Added start:dev script with hot reload support using --hot flag
- Added start:prod script for production execution
- Removed TODO comments from JournalRepository methods
- Added blank lines between updateEntry and deleteEntry methods for improved code formatting

Rafael-Sapalo committed:
feat: add NODE_ENV to journal service scripts and ignore Drizzle migrations directory

- Added NODE_ENV=development to start:dev script for development environment configuration
- Added NODE_ENV=production to start:prod script for production environment configuration
- Added .drizzle/ directory to .gitignore to exclude Drizzle migration artifacts
- Removed trailing newline from .gitignore file

Rafael-Sapalo committed:
feat: add Drizzle configuration, database scripts, and request validation to journal service

- Created drizzle.config.ts with PostgreSQL configuration using environment variables and SSL support
- Added database management scripts: db:generate, db:migrate, db:seed, db:push for schema operations
- Added docs:build script for TypeDoc API documentation generation
- Added @hono/standard-validator and zod dependencies for request validation
- Created journal.zod.ts with createEntriySchema and updateEntr

Rafael-Sapalo committed:
feat: implement updateEntry method with transaction support and automatic timestamp updates

- Added $onUpdate hook to updatedAt field in journal_entries table to automatically set timestamp on updates
- Implemented updateEntry method in JournalRepository with database transaction support
- Added existence check for journal entry before update with error handling
- Added conditional update logic to only update changed fields (title, content, mood)
- Added early return if no fields changed to avoid unnecessary database operations
- Adde

Rafael-Sapalo committed:
feat: add journal-service to Docker Compose configuration with database and messaging support

- Added JOURNAL_SERVICE_URL environment variable to gateway service pointing to http://journal-service:3003
- Created journal-service container with build context from ./services/journal directory
- Exposed port 3003 for journal service with port mapping
- Added environment variables for gateway URL, Redpanda messaging, and PostgreSQL database connection
- Added service dependencies on Redpanda and PostgreSQL with

Rafael-Sapalo committed:
feat: add await keyword to updateEntry transaction and fix whitespace formatting

- Added await keyword to db.transaction call in updateEntry method for proper async handling
- Fixed trailing whitespace on multiple lines within the transaction block
- Improved code consistency with proper spacing after conditional blocks

Mord0rak commented on elyrii_server/services/journal/index.ts:
```diff
@@ -0,0 +1,15 @@
+
+const journalService = new JournalService();
+
+Bun.serve({
```
NON

Rafael-Sapalo replied:
Comment ça non ?

Mord0rak reviewed (changes requested):
de l'erreur handling par pitie

Rafael-Sapalo committed:
feat: add error handler to journal service with logging and 500 response

- Added error callback to Bun.serve configuration for centralized error handling
- Added console.error logging with "[Journal Service Error]" prefix for error tracking
- Added generic "Internal Server Error" response with 500 status code for unhandled errors

Rafael-Sapalo committed:
feat: add Dockerfile for journal service and remove Docker ignore rule

- Removed Dockerfile from .gitignore to allow service-specific Docker configurations
- Created Dockerfile for journal service using oven/bun:1 base image
- Added package installation step with bun install using package.json and bun.lock
- Exposed port 3003 for journal service
- Set entrypoint to run start:prod script for production deployment
